routing/pathfind_test.go

```diff
@@ -668,20 +692,23 @@ func testBasicGraphPathFindingCase(t *testing.T, graph *channeldb.ChannelGraph,
 	target := aliases[test.target]
 	path, err := findPath(
 		nil, graph, nil, sourceNode, target, ignoredVertexes,
-		ignoredEdges, paymentAmt, test.feeLimit, nil,
+		ignoredEdges, paymentAmt, test.pathRestrictions, nil, finalHopCLTV,
 	)
-	if test.expectFailureNoPath {
-		if err == nil {
-			t.Fatal("expected no path to be found")
-		}
-		return
-	}
+
 	if err != nil {
-		t.Fatalf("unable to find path: %v", err)
+
+		if IsError(err, test.expectedFailure) {
```

Comment:
If think the default expectedFailure (if not specified in the testcase) is ErrNoPathFound. So what happens here is that if an unexpected error occurs and it happens to be ErrNoPathFound, the test will pass?